docs/source/models/concept_models/vivarium_census_synthdata/concept_model.rst

```diff
@@ -2807,6 +2807,349 @@ have not been included. These are:
+    - 1
+    - "Borrowed" SSN, Copy from within Household, Numeric miswriting 
+    - Note that not all types of noise apply to all observers, details below 
+  * - ITIN
+    - Taxes 1040
+    - 1%
+    - 1
+    - Copy from within Household, Numeric miswriting 
+    - Note that not all types of noise apply to all observers 
+  * - Income / Wages
+    - Taxes (both)
+    - 1%
+    - 1
+    - Numeric miswriting
+    - Note that wages and income are on separate tax forms and noise is applied to each separately 
+  * - Employer ID 
+    - Taxes (both)
+    - 1%
+    - 1
+    - Numeric miswriting
+    - 
+  * - Employer Name 
+    - Taxes (both)
+    - 1%
+    - 1
+    - OCR, typographic 
+    - 
+  * - Type of Tax Form  
+    - Taxes (both)
+    - 1%
+    - N/A
+    - Incorrect select 
+    - 
+  * - Type of SSA Event 
+    - SSA 
+    - 1%
+    - N/A
+    - Incorrect select 
+    - 
+  * - Date of SSA Event 
+    - SSA 
+    - 1%
+    - N/A
+    - Numeric miswriting, month and day swap 
+    - 
+
+The below section further describes types of noise including any code 
+available and information for implementation. **Software engineering team - please alert the research team if any of the below looks to be particularly challenging for further discussion.**
+
+**Converting from Noise Inputs to Noise Function Parameters for OCR, Phonetic, Typographic and Numeric Miswriting** 
+
+For simplicity for the end user, they will input two values for noise level: 
+
+#. Rows with Errors 
+#. Errors per Row 
+
+For types of noise others than those listed above, the only parameter is rows with error 
+and can be followed exactly. For noise types above, the functions we will use take different parameters: 
+
+#. Rows Eligible for Errors 
+#. Error Rate per Character OR per Token 
+
+These parameters, while related are different in key ways and therefore we need to 
+provide a way to convert between them. 
+
+Starting with #1 from both lists, we will calculate the number of rows eligible 
+for errors to approximate the user inputted rows with errors given. To calculate 
+this we can: 
+
+#. Compute the average length of strings in the column 
+#. Probability of at least 1 error = 1 - ((1 - (user defined errors per row / average string length))^(average string length))
+#. Rows eligible for errors = User defined rows with errors / probability of at least 1 error 
+
+Next, once a row is selected to receive noise, we define how much noise it should receive. 
+The default value here is 1 error per row, but can be changed. To calculate the error rate 
+per character/token, we can use this equation: 
+
+Error rate per Character/Token = (user defined errors per row)/(average string length)
+
+This approach has a number of assumptions and limitations: 
+
+- We assume that there are a similar number of tokens and characters in each string. This is approximately true for longer strings. While it is incorrect for short strings, we feel it is still reasonable overall. 
```

Comment:
I don't think it's true that the number of tokens is approximately equal to the number of characters for long strings. It's plausible that the ratio of tokens to characters could approach some ratio as the strings get longer, but that's not even clear because it depends on what type of data the strings represent (e.g., the ratio for names would probably be different from the ratios for addresses, birthdays, or zipcodes).

Reply:
Ah, you are right; I was misunderstanding what "tokens" really meant (I thought they were dense, i.e. every 3-character string was a token).

This can be solved pretty simply: sample 100 values and calculate the number of tokens in each, then take the average.